libcloudforensics/aws.py

```diff
@@ -0,0 +1,726 @@
+    """Create an AWS resource object.
+
+    Attributes:
+      service (str): The AWS service to use.
+      region (str): The region is which to create new resources in.
+
+    Returns:
+      boto3.Session.Resource: An AWS EC2 resource object.
+    """
+    if region:
+      return CreateSession().resource(
+          service_name=service, region_name=region
+      )
+    return CreateSession().resource(
+        service_name=service, region_name=self.default_region
+    )
+
+  def ListInstances(self, region=None, filters=None):
+    """List instances of an AWS account.
+
+    Attributes:
+      region (str): The region from which to list instances.
+      filters (list(dict)): Filters for the query.
+
+    Returns:
+      dict: Dictionary with name and metadata for each instance.
+
+    Example usage:
+      ListInstances(region='us-east-1', filters=[
+        dict(Name='instance-id', Values=['a-particular-instance-id'])
+      ])
+    """
+    if not filters:
+      filters = []
+
+    instances = dict()
+    have_all_tokens = False
+    next_token = None
+
+    while not have_all_tokens:
+      if next_token:
+        response = self.ClientApi(
+            EC2_SERVICE, region=region).describe_instances(
+                Filters=filters,
+                NextToken=next_token)
+      else:
+        response = self.ClientApi(
+            EC2_SERVICE, region=region).describe_instances(
+                Filters=filters)
+
+      for reservation in response["Reservations"]:
+        for instance in reservation["Instances"]:
+          instance_name = None
+          if instance.get('Tags'):
+            for tag in instance['Tags']:
+              instance_name = tag.get('Value') if tag.get('Key') == 'Name' \
+                else None
+              if instance_name is not None:
+                break
+          # Terminated instances are filtered out
+          if instance['State']['Name'] != 'terminated':
+            instance_info = dict(region=self.default_region,
+                                 zone=instance['Placement']['AvailabilityZone'])
+            if instance_name:
+              instance_info.update(name=instance_name)
+            instances[instance['InstanceId']] = instance_info
+
+      next_token = response.get('NextToken')
+      if not next_token:
+        have_all_tokens = True
+
+    return instances
+
+  def ListVolumes(self, region=None, filters=None):
+    """List volumes of an AWS account.
+
+    Attributes:
+      region (str): The region from which to list the volumes.
+      filters (list(dict)): Filter for the query.
+
+    Returns:
+      dict: Dictionary with name and metadata for each volume.
+
+    Example usage:
+      # List volumes attached to the instance 'a-particular-instance-id'
+      ListVolumes(filters=[
+        dict(Name='attachment.instance-id', Values=['a-particular-instance-id'])
+      ])
+    """
+    if not filters:
+      filters = []
+
+    volumes = dict()
```

Comment:
Thought for a future PR: it would be nice to make the `dict` creation call consistent. I see `dict()` and `{}` being used in the codebase

Reply:
This is not consistent in `gcp.py` either, I'll fix it there. I would rather go with the `dict()` call, wdyt?

Reply:
I'd go for the literal (`{}`, or `[]` if you're using lists), it's clearer than a function call, and according to the Internet, it's also a bit faster (but that's really bold coming from Python :P)